src/server/scripts/Commands/cs_debug.cpp

```diff
@@ -1045,7 +1045,8 @@ class debug_commandscript : public CommandScript
             return false;
 
         uint32 animId = atoi((char*)args);
-        handler->GetSession()->GetPlayer()->HandleEmoteCommand(animId);
+        if (Unit* unit = handler->getSelectedUnit())
+            unit->HandleEmoteCommand(animId);
```

Comment:
Perhaps this can be changed to

```
        Unit* unit = handler->getSelectedUnit();
        if (!unit)
            unit = handler->GetSession()->GetPlayer();
        unit->HandleEmoteCommand(animId);
```

to keep the old behaviour and support the new target as well?